saleor/graphql/order/tests/test_order_grant_refund_update.py

```diff
@@ -0,0 +1,420 @@
+        == amount
+    )
+    assert granted_refund.reason == current_reason
+
+
+def test_grant_refund_update_only_reason_by_user(
+    staff_api_client, app, permission_manage_orders, order
+):
+    # given
+    current_reason = "Granted refund reason."
+    current_amount = Decimal("10.00")
+    granted_refund = order.granted_refunds.create(
+        amount_value=current_amount,
+        currency=order.currency,
+        reason=current_reason,
+        app=app,
+    )
+    granted_refund_id = to_global_id_or_none(granted_refund)
+    staff_api_client.user.user_permissions.add(permission_manage_orders)
+    reason = "new reason"
+    variables = {
+        "id": granted_refund_id,
+        "input": {"reason": reason},
+    }
+
+    # when
+    response = staff_api_client.post_graphql(ORDER_GRANT_REFUND_UPDATE, variables)
+
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["orderGrantRefundUpdate"]
+    assert not data["errors"]
+
+    assert len(data["order"]["grantedRefunds"]) == 1
+    granted_refund.refresh_from_db()
+    granted_refund_assigned_to_order = data["order"]["grantedRefunds"][0]
+    assert granted_refund_assigned_to_order == data["grantedRefund"]
+    assert granted_refund_assigned_to_order["reason"] == reason == granted_refund.reason
+    assert granted_refund.amount_value == current_amount
+
+
+def test_grant_refund_update_by_user_missing_permission(staff_api_client, app, order):
+    # given
+    current_reason = "Granted refund reason."
+    current_amount = Decimal("10.00")
+    granted_refund = order.granted_refunds.create(
+        amount_value=current_amount,
+        currency=order.currency,
+        reason=current_reason,
+        app=app,
+    )
+    amount = Decimal("10.00")
+    reason = "Granted refund reason."
+    variables = {
+        "id": to_global_id_or_none(granted_refund),
+        "input": {"amount": amount, "reason": reason},
+    }
+
+    # when
+    response = staff_api_client.post_graphql(ORDER_GRANT_REFUND_UPDATE, variables)
+
+    # then
+    assert_no_permission(response)
+
+
+def test_grant_refund_update_by_user_missing_input(
+    staff_api_client, staff_user, order, permission_manage_orders
+):
+    # given
+    current_reason = "Granted refund reason."
+    current_amount = Decimal("10.00")
+    granted_refund = order.granted_refunds.create(
+        amount_value=current_amount,
+        currency=order.currency,
+        reason=current_reason,
+        user=staff_user,
+    )
+
+    variables = {
+        "id": to_global_id_or_none(granted_refund),
+        "input": {},
+    }
+    staff_api_client.user.user_permissions.add(permission_manage_orders)
+
+    # when
+    response = staff_api_client.post_graphql(ORDER_GRANT_REFUND_UPDATE, variables)
+
+    # then
+    content = get_graphql_content_from_response(response)
+    errors = content["data"]["orderGrantRefundUpdate"]["errors"]
+    assert len(errors) == 1
+    assert errors[0]["code"] == "REQUIRED"
```

Comment:
We usually compare to the value from enum like:
```suggestion
    assert errors[0]["code"] == OrderGrandRefundUpdateErrorCode.REQUIRED.name
```